presto-main/src/test/java/com/facebook/presto/metadata/TestFunctionRegistry.java

```diff
@@ -132,6 +149,80 @@ public void testListingHiddenFunctions()
         assertFalse(names.contains("like"), "Expected function names " + names + " not to contain 'like'");
     }
 
+    @Test
+    public void testCanCoerce()
+    {
+        assertTrue(canCoerce(BIGINT, BIGINT));
+        assertTrue(canCoerce(UNKNOWN, BIGINT));
+        assertFalse(canCoerce(BIGINT, UNKNOWN));
+
+        assertTrue(canCoerce(BIGINT, DOUBLE));
+        assertTrue(canCoerce(DATE, TIMESTAMP));
+        assertTrue(canCoerce(DATE, TIMESTAMP_WITH_TIME_ZONE));
+        assertTrue(canCoerce(TIME, TIME_WITH_TIME_ZONE));
+        assertTrue(canCoerce(TIMESTAMP, TIMESTAMP_WITH_TIME_ZONE));
+        assertTrue(canCoerce(VARCHAR, REGEXP));
+        assertTrue(canCoerce(VARCHAR, LIKE_PATTERN));
+        assertTrue(canCoerce(VARCHAR, JSON_PATH));
+
+        assertFalse(canCoerce(DOUBLE, BIGINT));
+        assertFalse(canCoerce(TIMESTAMP, TIME_WITH_TIME_ZONE));
+        assertFalse(canCoerce(TIMESTAMP_WITH_TIME_ZONE, TIMESTAMP));
+        assertFalse(canCoerce(VARBINARY, VARCHAR));
+
+        assertTrue(canCoerce(UNKNOWN.getTypeSignature(), parseTypeSignature("array<bigint>")));
+        assertFalse(canCoerce(parseTypeSignature("array<bigint>"), UNKNOWN.getTypeSignature()));
+        assertTrue(canCoerce(parseTypeSignature("array<bigint>"), parseTypeSignature("array<double>")));
+        assertFalse(canCoerce(parseTypeSignature("array<double>"), parseTypeSignature("array<bigint>")));
+        assertTrue(canCoerce(parseTypeSignature("map<bigint,double>"), parseTypeSignature("map<bigint,double>")));
+        assertFalse(canCoerce(parseTypeSignature("map<bigint,double>"), parseTypeSignature("map<double,double>"))); // map covariant cast is not supported yet
+        assertTrue(canCoerce(parseTypeSignature("row<bigint,double,varchar>"), parseTypeSignature("row<bigint,double,varchar>")));
+        assertFalse(canCoerce(parseTypeSignature("row<bigint,double,varchar>('a','b','c')"), parseTypeSignature("row<bigint,double,varchar>")));
+        assertFalse(canCoerce(parseTypeSignature("row<bigint,double,varchar>"), parseTypeSignature("row<bigint,double,varchar>('a','b','c')")));
+        assertTrue(canCoerce(parseTypeSignature("row<bigint,double,varchar>('a','b','c')"), parseTypeSignature("row<bigint,double,varchar>('a','b','c')")));
+    }
+
+    @Test
+    public void testGetCommonSuperType()
+    {
+        assertCommonSuperType(BIGINT, BIGINT, BIGINT);
+        assertCommonSuperType(UNKNOWN, BIGINT, BIGINT);
```

Comment:
In Presto, UNKNOWN means anything. For example, `SELECT null UNION SELECT 1` should be `BIGINT`.